spacelift/data_contexts.go

```diff
@@ -86,18 +103,57 @@ func dataContextsRead(ctx context.Context, d *schema.ResourceData, meta interfac
 	return nil
 }
 
-func flattenDataContextsList(contexts []*structs.Context) []map[string]interface{} {
-	wps := make([]map[string]interface{}, len(contexts))
+func flattenDataContextsList(contexts []*structs.Context, labelFilters [][]string) []map[string]interface{} {
+	wps := []map[string]interface{}{}
+
+	for _, ctx := range contexts {
+		if !matchesLabels(ctx.Labels, labelFilters) {
+			continue
+		}
 
-	for index, ctx := range contexts {
-		wps[index] = map[string]interface{}{
+		wps = append(wps, map[string]interface{}{
 			"context_id":  ctx.ID,
 			"description": ctx.Description,
 			"labels":      ctx.Labels,
 			"name":        ctx.Name,
 			"space_id":    ctx.Space,
-		}
+		})
 	}
 
 	return wps
 }
+
+func matchesLabels(labelSet []string, labelFilters [][]string) bool {
+	if len(labelFilters) == 0 {
+		return true
+	}
+
+	for _, labelFilter := range labelFilters {
+		if len(labelFilter) == 0 {
+			continue
+		}
+
+		var matchesFilter bool
+		for _, label := range labelFilter {
+			if contains(labelSet, label) {
+				matchesFilter = true
+			}
+		}
+
+		if !matchesFilter {
+			return false
+		}
+	}
+
+	return true
+}
+
+func contains(set []string, value string) bool {
```

Comment:
You could use https://pkg.go.dev/golang.org/x/exp/slices#Contains here

Reply:
Oh wow, will do!